DetoxSync/DetoxSync/Utils/_DTXTimerTrampoline.m

```diff
@@ -112,47 +112,24 @@ - (void)setDisplayLink:(CADisplayLink*)displayLink
 #endif
 }
 
-- (void)fire:(id)timer
+- (void)fire
 {
-	//This is to ensure the timer is still valid after fire.
-	CFRunLoopRef runloop = CFRunLoopGetCurrent();
-	CFRunLoopMode mode = CFRunLoopCopyCurrentMode(runloop);
-	CFRunLoopPerformBlock(runloop, mode, ^{
-		if(CFRunLoopTimerIsValid((__bridge CFRunLoopTimerRef)timer) == NO)
-		{
-			[self untrack];
-			
-			CFRelease(mode);
-			
-			return;
-		}
-		
-		CFRunLoopPerformBlock(runloop, mode, ^{
-			if(CFRunLoopTimerIsValid((__bridge CFRunLoopTimerRef)timer) == NO)
-			{
-				[self untrack];
-				
-				CFRelease(mode);
-				
-				return;
-			}
-			
-			CFRelease(mode);
-		});
-	});
-	
 	if(_callback)
 	{
 		CFRunLoopTimerContext ctx;
-		CFRunLoopTimerGetContext((__bridge CFRunLoopTimerRef)timer, &ctx);
-		_callback((__bridge CFRunLoopTimerRef)timer, ctx.info);
-		return;
+		CFRunLoopTimerGetContext((__bridge CFRunLoopTimerRef)_timer, &ctx);
+		_callback((__bridge CFRunLoopTimerRef)_timer, ctx.info);
+	}
+
+	if(_target && _sel) {
+		IMP impl = [_target methodForSelector:_sel];
+		void (*func)(id, SEL, NSTimer*) = (void *)impl;
+		func(_target, _sel, _timer);
+	}
```

Comment:
This is a small tidy-up recommended by https://stackoverflow.com/a/20058585

Reply:
:+1: